src/components/product.js

```diff
@@ -111,73 +114,62 @@ export default class Product extends Component {
    * @return {Object} image object.
    */
   get image() {
-    if (!this.model.selectedVariant || !this.model.selectedVariant.image || !this.model.selectedVariant.image.variants.length) {
+    if (!this.selectedVariant || (!this.selectedVariant.image && !this.options.contents.imgWithCarousel)) {
       return null;
     }
-    const availableSizes = this.model.selectedVariant.image.variants;
-
-    let imageSize = 'grande';
-
-    if (this.options.width && this.options.layout === 'vertical') {
-      imageSize = availableSizes.filter((image) => {
-        const containerWidth = parseInt(this.options.width, 10);
-        return parseInt(image.dimension, 10) >= containerWidth * 1.5;
-      })[0].name;
-    }
+    const imageSize = parseInt(this.options.width, 10) || 480;
+    let image;
+    let id;
 
-    if (this.options.imageSize) {
-      imageSize = this.options.imageSize;
-    }
+    const imageOptions = {
+      maxWidth: imageSize,
+      maxHeight: imageSize,
+    };
 
-    let sourceImage = this.model.selectedVariant.image;
     if (this.selectedImage) {
-      sourceImage = this.model.images.filter((image) => {
-        return image.id === this.selectedImage.id;
-      })[0];
+      id = this.selectedImage.id;
+      image = ShopifyBuy.Image.Helpers.imageForSize(this.selectedImage, imageOptions);
```

Comment:
would this var be  better named as `src`? Elsewhere, `image` refers to a hash with a `src` attribute. Here, `image` refers to a literally a `src` which is returned in the statement below.